src/sass/_functions.scss

```diff
@@ -229,3 +229,40 @@ Will generate for example following classes:
         @include createSelector(#{$newSelector}, $properties);
     }
 }
+
+// @include breakpoint('xs') { width: 90%; }
+@mixin breakpoint($bpName, $breakpoints: $breakpoints) {
+    $bpValue: map-get($breakpoints, $bpName);
+    @media (min-width: $bpValue) { @content; }
+}
+
+@mixin createClasses($map) {
+    @each $selector, $definition in $map {
+        .#{$selector} {
+            @each $property, $value in $definition {
+                #{$property}: $value;
+            }
+        }
+    }
+}
+
+@mixin responsiveClasses($selector, $_breakpoints: $breakpoints) {
+    @each $name, $value in $breakpoints {
+        $responsiveSelector: 'r-' + $name + '-' + $selector;
+        .#{$responsiveSelector} {
+            @include breakpoint($name, $breakpoints) {
+                @content;
+            }
+        }
+    }
+}
+
+@mixin createResponsiveClasses($map) {
+    @each $selector, $definition in $map {
+        @include responsiveClasses($selector) {
+            @each $property, $value in $definition {
```

Comment:
Wait a minute. This happens multiple times on the codebase. Let's make it a mixin and call that from everywhere.

```
@mixin properties($properties) {
  @each $property, $value in $properties {
    #{$property}: #{$value};
  }
}
```